### PR TITLE
Add http_auth support for curator

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
Add http_auth support for curator. Needed for gardener/gardener#851 .

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
 
**Release note**:

```improvement operator
Curator now supports `http_auth` option for basic auth against elasticsearch.
```

### CODE REVIEW

File: curator-es/main.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+
+	"github.com/gardener/logging/curator-es/pkg/config"
+	curator "github.com/gardener/logging/curator-es/pkg/curator"
```
Comment (vpnachev) [inline curator-es/main.go]: There is no need to rename the package with the same name.

File: curator-es/main.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+func main() {
+	configPath := flag.String("config", "/etc/config/config.yml", "The config file for the curator")
+	esAPI := flag.String("es_api", "", "The Elasticsearch API in format <host>:<port>")
```
Comment (vpnachev) [inline curator-es/main.go]:
```suggestion
	esAPI := flag.String("es-api-server", "", "The Elasticsearch API server in format <host>:<port>")
```

File: curator-es/main.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	diskSpaceThreshold := flag.Int64("disk_threshold", 100000000, "The minimum maximum disk space left before deletion of the index")
```
Comment (vpnachev) [inline curator-es/main.go]:
```suggestion
	diskSpaceThreshold := flag.Int64("disk-threshold", 100000000, "The minimum maximum disk space left before deletion of the index")
```

File: curator-es/pkg/config/config.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		return nil, errors.New("No configuration path passed")
+	}
+
+	// Read the configuration file.
+	configRaw, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Marshal the configuration file into config struct.
```
Comment (vlvasilev) [inline curator-es/pkg/config/config.go]: Unmarshal

**Comment by ialidzhikov** [comment]
Changes applied, PTAL.

**Pull request merged**